catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/util/DigitalSignature.java

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.commands.util;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.InvalidKeyException;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.Signature;
+import java.security.SignatureException;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import org.apache.commons.io.IOUtils;
+import org.codice.ddf.security.common.Security;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DigitalSignature {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(DigitalSignature.class);
+
+  private static final int BUFFER_SIZE = 1024;
+
+  private static final int OFFSET = 0;
+
+  private KeyStore keyStore;
+
+  public DigitalSignature() {
+    this.keyStore = Security.getInstance().getSystemKeyStore();
+  }
+
+  public DigitalSignature(KeyStore keyStore) {
+    this.keyStore = keyStore;
+  }
+
+  public byte[] createDigitalSignature(InputStream data, String alias, String password)
+      throws IOException {
+    PrivateKey privateKey = getPrivateKey(alias, password);
+
+    if (privateKey == null) {
+      throw new CatalogCommandRuntimeException("Unable to retrieve private key");
```

Comment:
Since there isn't a way for someone running the command to see into DDF's system keystore there isn't much they can do if the keys we're using aren't being loaded into the keystore. 

In a future iteration we could make the private key and certificate configurable by the person running the command however that's probably out of scope.